### PR TITLE
Add data capture config to SageMaker Deployments

### PR DESCRIPTION
**Pull request by jonwiggins**
## Related Issues/PRs
None

## What changes are proposed in this pull request?

This adds `data_capture_config` as a mostly pass-through variable such that users can these settings when deploying models to AWS SageMaker.

Let me know if you need me to make any changes.

## How is this patch tested?

I deployed a few models using this patch, and saw that the data capture configs had been set.

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

This adds `data_capture_config` as a parameter in the SageMaker deployment functions such that users can these settings when deploying models to AWS SageMaker.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [X] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category">Notes</a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [X] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by dbczumar — COMMENTED** [review]
@jonwiggins Thank you for introducing this feature! The PR looks great! Can we also add this capability to `mlflow.sagemaker.SageMakerDeploymentClient.create_deployment()` and ``mlflow.sagemaker.SageMakerDeploymentClient.update_deployment()` by forwarding `config["data_capture_config"]` here https://github.com/mlflow/mlflow/blob/7d73598c99a1697b272f56a40ddecb37953347b5/mlflow/sagemaker/__init__.py#L2078-L2094 and here https://github.com/mlflow/mlflow/blob/7d73598c99a1697b272f56a40ddecb37953347b5/mlflow/sagemaker/__init__.py#L2293-L2309?

**Comment by jonwiggins** [comment]
@dbczumar Great thanks.

I tried to make the changes you requested.
Let me know what you think.
Happy to make any needed changed.

File: mlflow/sagemaker/__init__.py [hunk 1]
```diff
@@ -1919,7 +1948,7 @@ def create_deployment(self, name, model_uri, flavor=None, config=None, endpoint=
                        If ``None``, a flavor is automatically selected from the model's available
                        flavors. If the specified flavor is not present or not supported for
                        deployment, an exception will be thrown.
-        :param config: Configuration paramaters. The supported paramaters are:
```
Comment (dbczumar) [inline mlflow/sagemaker/__init__.py:1922]: Thanks for fixing these typos too :)

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks so much for your contribution, @jonwiggins !

**Pull request merged**